docs/files/integration.rst

```diff
@@ -0,0 +1,162 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2019 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+.. _integration:
+
+Integrating Files
+=================
+
+Invenio provides a bundle of modules to accommodate all around needs about
+file management. You can install all the necessary modules at once, or due to
+invenio modular architecture you can hand pick only those you need.
+
+In case you have scaffolded your invenio application with
+`cookiecutter-invenio-instance <https://github.com/inveniosoftware/cookiecutter-invenio-instance>`_
+files bundle is already a part of your solution, the only step you need to
+take is to enable them through configuration.
+
+In case you want all the modules automatically installed you can navigate to
+your project and with your virtual environment selected you can
+
+.. code-block:: console
+
+   $ pip install invenio[auth]
+
+
+Files bundle
+------------
+
+As mentioned before functionality for files support is broken down in different
+parts to provide as much flexibility as possible.
+
+`invenio-files-rest <https://invenio-files-rest.readthedocs.io/>`_
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: console
+
+   $ pip install invenio-files-rest
+
+:code:`invenio-files-rest` is the first required module for managing files
+with your application, and some of its key functions is to allow you to store
+and retrieve files in a similar way to Amazon S3 APIs.
+
+- Configurable files storage
+- Secure REST APIs for upload/download
+- Support for large file uploads and multipart upload.
+- File integrity monitoring
+- Customizable access control
+
+After you install the module on your environment you just have to register
+the invenio-files-rest blueprint in the list of blueprints defined in the
+:code:`entry_points` of the :code:`setup.py` of your application.
+
+.. code-block:: python
+
+    entry_points={
+        "invenio_app.api_blueprints": [
+            "invenio_files = invenio_files_rest.views:blueprint",
```

Comment:
Was this change needed when you were testing? In the end for my upgrade I didn't have to do this.